src/ospjolt/activescene/joltinteg.h

```diff
@@ -0,0 +1,306 @@
+			return false;
+		}
+	}
+};
+/**
+ * @brief The different broad phase layers (currently identical to physics Layers, but might change in the future)
+ */
+namespace BroadPhaseLayers
+{
+	static constexpr BroadPhaseLayer NON_MOVING(0);
+	static constexpr BroadPhaseLayer MOVING(1);
+	static constexpr uint NUM_LAYERS(2);
+};
+
+/**
+ * @brief Class that defines a mapping between object and broadphase layers.
+ */
+class BPLayerInterfaceImpl final : public BroadPhaseLayerInterface
+{
+public:
+									BPLayerInterfaceImpl()
+	{
+		// Create a mapping table from object to broad phase layer
+		mObjectToBroadPhase[Layers::NON_MOVING] = BroadPhaseLayers::NON_MOVING;
+		mObjectToBroadPhase[Layers::MOVING] = BroadPhaseLayers::MOVING;
+	}
+
+	virtual uint					GetNumBroadPhaseLayers() const override
+	{
+		return BroadPhaseLayers::NUM_LAYERS;
+	}
+
+	virtual BroadPhaseLayer			GetBroadPhaseLayer(ObjectLayer inLayer) const override
+	{
+		JPH_ASSERT(inLayer < Layers::NUM_LAYERS);
+		return mObjectToBroadPhase[inLayer];
+	}
+
+#if defined(JPH_EXTERNAL_PROFILE) || defined(JPH_PROFILE_ENABLED)
+	/// Get the user readable name of a broadphase layer (debugging purposes)
+	virtual const char *			GetBroadPhaseLayerName(BroadPhaseLayer inLayer) const override
+	{
+		if (inLayer == BroadPhaseLayers::NON_MOVING) {
+			return "NON_MOVING";
+		}
+		else {
+			return "MOVING";
+		}
+	}
+#endif // JPH_EXTERNAL_PROFILE || JPH_PROFILE_ENABLED
+
+private:
+	BroadPhaseLayer					mObjectToBroadPhase[Layers::NUM_LAYERS];
+};
+/**
+ * @brief Class that determines if an object layer can collide with a broadphase layer
+ */
+class ObjectVsBroadPhaseLayerFilterImpl : public ObjectVsBroadPhaseLayerFilter
+{
+public:
+	virtual bool				ShouldCollide(ObjectLayer inLayer1, BroadPhaseLayer inLayer2) const override
+	{
+		switch (inLayer1)
+		{
+		case Layers::NON_MOVING:
+			return inLayer2 == BroadPhaseLayers::MOVING;
+		case Layers::MOVING:
+			return true;
+		default:
+			JPH_ASSERT(false);
+			return false;
+		}
+	}
+};
+
+//Forward declaration
+struct ACtxJoltWorld;
+
+//The physics callback to sync bodies between osp and jolt
+class PhysicsStepListenerImpl : public PhysicsStepListener
+{
+public:
+    PhysicsStepListenerImpl(ACtxJoltWorld* pContext) : m_context(pContext) {};
+    virtual void OnStep(float inDeltaTime, PhysicsSystem& inPhysicsSystem) override;
+
+private:
+    ACtxJoltWorld* m_context;
+};
+
+using TransformedShapePtr_t = std::unique_ptr<TransformedShape>;
+using ShapeStorage_t = osp::Storage_t<osp::active::ActiveEnt, TransformedShapePtr_t>;
+
+/**
+ * @brief Represents an instance of a Jolt physics world in the scene
+ */
+struct ACtxJoltWorld
+{
+
+    struct ForceFactorFunc
+    {
+        using UserData_t = std::array<void*, 6u>;
+        using Func_t = void (*)(JoltBodyId joltBodyId, OspBodyId ospBodyId, ACtxJoltWorld const&, UserData_t, osp::Vector3&, osp::Vector3&) noexcept;
+
+        Func_t      m_func;
+        UserData_t  m_userData;
```

Comment:
Note that neither of these member variables will have any initialization, so if they are read before something sets them, the read values will be arbitrary.